executor/poolmgr/gp.go

```diff
@@ -228,21 +226,14 @@ func (gp *GenericPool) _choosePod(newLabels map[string]string) (*apiv1.Pod, erro
 		for i := range podList.Items {
 			pod := podList.Items[i]
 
-			// If a pod has no IP it's not ready
-			if len(pod.Status.PodIP) == 0 || string(pod.Status.Phase) != POD_PHASE_RUNNING {
+			// If a pod has no IP or any of pod's containers
+			// is not in ready state, it's not ready.
+			if len(pod.Status.PodIP) == 0 || !fission.IsReadyPod(&pod) {
```

Comment:
Shouldn't the PodIP check be in the `IsReadyPod` function if it is part of the decision of whether a pod is ready?

Reply:
I think we can remove pod ip check here since a pod definitely has IP when its ready